libs/coin-modules/coin-ton/src/synchronisation.ts

```diff
@@ -24,43 +47,134 @@ export const getAccountShape: GetAccountShape<Account> = async info => {
+      newJettonTxs.push(...tmpJettonTxs);
     } else {
-      // if they are the same, we have no new ops
+      // if they are the same, we have no new ops (including tokens)
       if (oldOps[0].extra.lt !== last_transaction_lt) {
-        const [tmpTxs] = await Promise.all([getTransactions(address, oldOps[0].extra.lt)]);
+        const [tmpTxs, tmpJettonTxs] = await Promise.all([
+          getTransactions(address, oldOps[0].extra.lt),
+          getJettonTransfers(address, oldOps[0].extra.lt),
+        ]);
         newTxs.transactions.push(...tmpTxs.transactions);
         newTxs.address_book = { ...newTxs.address_book, ...tmpTxs.address_book };
+        newJettonTxs.push(...tmpJettonTxs);
       }
     }
   }
 
   const newOps = flatMap(newTxs.transactions, mapTxToOps(accountId, address, newTxs.address_book));
-  const operations = mergeOps(oldOps, newOps);
+  const newJettonOps = flatMap(
+    newJettonTxs,
+    mapJettonTxToOps(accountId, address, newTxs.address_book),
+  );
+  const operations = shouldSyncFromScratch ? newOps : mergeOps(oldOps, newOps);
+  const subAccounts = await getSubAccounts(
+    info,
+    accountId,
+    newJettonOps,
+    blacklistedTokenIds,
+    shouldSyncFromScratch,
+  );
 
   const toReturn = {
     id: accountId,
     balance: new BigNumber(balance),
     spendableBalance: new BigNumber(balance),
     operations,
     operationsCount: operations.length,
+    subAccounts,
     blockHeight,
     xpub: publicKey,
     lastSyncDate: new Date(),
   } as Partial<Account>;
   return toReturn;
 };
 
+const getSubAccountShape = async (
+  info: AccountShapeInfo,
+  parentId: string,
+  token: TokenCurrency,
+  ops: TonOperation[],
+  shouldSyncFromScratch: boolean,
+): Promise<Partial<SubAccount>> => {
+  const tokenAccountId = encodeTokenAccountId(parentId, token);
+  const walletsInfo = await fetchJettonWallets({
+    address: info.address,
+    jettonMaster: token.contractAddress,
+  });
+  if (walletsInfo.length !== 1) throw new Error("[ton] unexpected api response");
+  const { balance, address: jettonWalletAddress } = walletsInfo[0];
+  const oldOps = info.initialAccount?.subAccounts?.find(a => a.id === tokenAccountId)?.operations;
+  const operations = !oldOps || shouldSyncFromScratch ? ops : mergeOps(oldOps, ops);
+  const maybeExistingSubAccount =
+    info.initialAccount &&
+    info.initialAccount.subAccounts &&
+    info.initialAccount.subAccounts.find(a => a.id === tokenAccountId);
+
+  return {
+    type: "TokenAccount",
+    id: tokenAccountId,
+    parentId,
+    token: { ...token, contractAddress: jettonWalletAddress }, // the contract address is replaced for the jetton wallet address, it will be use for the token transfer
+    balance: new BigNumber(balance),
+    spendableBalance: new BigNumber(balance),
+    operations,
+    operationsCount: operations.length,
+    pendingOperations: maybeExistingSubAccount ? maybeExistingSubAccount.pendingOperations : [],
```

Comment:
I think we don't use pendingOperations in getAccountShape. 
operations with blockheight = null are considered as pending operations

you can just 

```
pendingOperations: [],
```
I think

Reply:
I tested and it didn't work as expected if I change the code. When the account is synchronized again (and the transaction wasn't received yet), the pending operation is no longer displayed.
We receive the pendingOperations in getAccountShape and we use them in postSync too.